doc/configuration.md

```diff
@@ -0,0 +1,247 @@
+Configuration
+=============
+The Datadog tracer's configuration can be specified in multiple ways:
+
+- programmatically in C++ code via the `TracerOptions` object defined in
+  [datadog/opentracing.h][1],
+- process-wide via setting values for certain [environment variables][4],
+- [dynamically][3] by reading JSON-formatted text, as in done in the [nginx
+  plugin][2].
+
+Most options support all three methods of configuration.
+
+Environment variables override any corresponding configuration in
+`TracerOptions` or loaded from JSON.
+
+Options
+-------
+### Agent Host
+The name of the host at which the Datadog Agent can be contacted.
+
+- **TracerOptions member**: `std::string agent_host`
+- **JSON property**: `"agent_host"` _string_
+- **Environment variable**: `DD_AGENT_HOST`
+- **Default value**: `"localhost"`
+
+### Agent Port
+The port on which the Datadog Agent is listening.
+
+- **TracerOptions member**: `uint32_t agent_port`
+- **JSON property**: `"agent_port"` _integer_
+- **Environment variable**: `DD_TRACE_AGENT_PORT`
+- **Default value**: `8126`
+
+### Agent URL
+As an alternative to specifying a host and port separately, a URL may be
+specified indicating where the Datadog Agent can be contacted.  Both TCP
+and Unix domain sockets are supported.  For more information about using a
+Unix domain socket, see the [relevant example][5].
+
+The following forms are supported:
+
+- `http://host` (TCP)
+- `http://host:port` (TCP)
+- `https://host` (TCP; TLS is not used)
+- `https://host:port` (TCP; TLS is not used)
+- `unix://path` (Unix domain socket)
+- `path` (Unix domain socket)
+
+- **TracerOptions member**: `std::string agent_url`
+- **JSON property**: `"agent_url"` _(string)_
+- **Environment variable**: `DD_TRACE_AGENT_URL`
+- **Default value**: `""`
+
+### Service Name
+The default service name to associate with spans produced by the tracer.
+Service name can be overridden programmatically on a per-span basis via the
+`datadog::opentracing::Span::setServiceName` member function.
+
+- **TracerOptions member**: `std::string service`
+- **JSON property**: `"service"` _(string)_
+- **Environment variable**: `DD_SERVICE`
+- **Required**
+
+### Service Type
+The default "service type" to associate with spans produced by the tracer.
+
+Service type is used in multiple places throughout Datadog to distinguish
+different categories of instrumented service from each other.  For example,
+it is used in the following ways:
+
+- to identify whether the service's spans need to be obfuscated
+- to control display of the service in the Datadog UI.
+
+Example values for service type are `web`, `db`, and `lambda`.
+
+- **TracerOptions member**: `std::string type`
+- **JSON property**: `"type"` _(string)_
+- **Default value**: `"web"`
+
+### Environment
+The default release environment in which the service is running, e.g. "staging"
```

Comment:
I'd usually mention prod first, eg: prod, dev or staging.
It's a fairly important thing to set, so maybe a bit more detail here so it sounds/feels as important as it is.
It's best to align the value for related host monitoring and metrics collection.

Reply:
I'll change the order.

I'm not familiar with all of the ways that `environment` affects things. I'll read the public docs, and add key points here, or at least link to the docs.